src/transformers/modeling_utils.py

```diff
@@ -3970,10 +4000,18 @@ def _find_mismatched_keys(
                         model_key in model_state_dict
                         and state_dict[checkpoint_key].shape != model_state_dict[model_key].shape
                     ):
-                        mismatched_keys.append(
-                            (checkpoint_key, state_dict[checkpoint_key].shape, model_state_dict[model_key].shape)
-                        )
-                        del state_dict[checkpoint_key]
+                        if (
+                            state_dict[checkpoint_key].shape[-1] == 1
+                            and state_dict[checkpoint_key].numel() * 2 == model_state_dict[model_key].numel()
+                        ):
+                            # Such mismatched weights are OK for 4bit quantizations.
+                            # Need more reliable condition here, ideally based on type(module)
```

Comment:
I don't know how reliable it is

Reply:
I thought about it but could not find a better way to verify 4bit param type:
- param object type is not available inside `_find_mismatched_keys()`
- `checkpoint_key` name is generic and it gives no clue whether it is 4-bit param.
- 8bit quant has no such problem because there are 8-bit torch types (unlike 4bit so far)
- using `ignore_mismatched_sizes` seems excessive for this case. 

could you suggest something?

Reply:
That's fine then! Could you just add a quick note on how these conditions help us to find the 4-bit param type, especially the second condition.

Reply:
edited the comment to explain reasons for the exception.
rebased to fresh main.